internal/controllers/nmc_reconciler.go

```diff
@@ -178,6 +178,10 @@ func (r *NMCReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 		errs = append(errs, fmt.Errorf("failed to GC in-use labels for NMC %s: %v", req.NamespacedName, err))
 	}
 
+	if err := r.helper.GarbageCollectWorkerPods(ctx, &nmcObj); err != nil {
```

Comment:
I am wondering if it won't be better to add a finalizer to the NMC object to make sure it is not being deleted before the worker pods are and if the NMC is being delete we go over all its worker pod and make sure we remove all failed pods instead of garbage collecting them a-syncronically.

 WDYT @yevgeny-shnaidman @TomerNewman